tools/convert_to_h5.cwl

```diff
@@ -1,44 +1,62 @@
 cwlVersion: v1.2
 class: CommandLineTool
 id: convert-to-h5
-doc: "Convert counts dir from STAR Solo to h5 file"
+doc: "Convert counts dir from STAR Solo to h5 file. If user wishes to use a mutli-map count raw file from the results, use raw_count_choice"
 
 requirements:
   - class: ShellCommandRequirement
   - class: DockerRequirement
-    dockerPull: "pgc-images.sbgenomics.com/brownm28/soupx:1.6.2"
+    dockerPull: "pgc-images.sbgenomics.com/d3b-bixu/soupx_r:1.6.2"
   - class: InlineJavascriptRequirement
   - class: InitialWorkDirRequirement
-    listing:
+    listing: 
+      - $(inputs.solo_counts_dir)
       - entryname: convert_to_h5.R
         entry:
           $include: ../scripts/convert_to_h5.R
+
   - class: ResourceRequirement
     coresMin: 2
     ramMin: 4000
 
 baseCommand: []
 arguments:
   - position: 0
+    shellQuote: false
+    valueFrom: >-
+      ${
+        var raw_dir = inputs.solo_counts_dir.path + "/" + inputs.soloFeatures + "/raw/";
+        var setup_raw_cmd = "mkdir RAW && cp " + raw_dir + "*.tsv RAW/ &&";
+        if (inputs.raw_count_choice == "Unique" || inputs.raw_count_choice == null){
+          setup_raw_cmd +=  " cp " + raw_dir + "matrix.mtx RAW/ &&"
+        }
+        else{
+          setup_raw_cmd += " cp " + raw_dir + "UniqueAndMult-" + inputs.raw_count_choice + ".mtx RAW/matrix.mtx &&";
+        }
+        return setup_raw_cmd;
+      }
```

Comment:
Just having fun
```suggestion
    valueFrom: >-
      mkdir RAW
      && cp $(inputs.solo_counts_dir.path)/$(inputs.soloFeatures)/raw/*.tsv RAW/ 
      && cp $(inputs.solo_counts_dir.path)/$(inputs.soloFeatures)/raw/$(["Unique",null].indexOf(inputs.raw_count_choice) != -1 ? "matrix" : "UniqueAndMult-" + inputs.raw_count_choice).mtx RAW/matrix.mtx &&
```

Reply:
Hmmm, I'll try it out tomorrow morning. I think I am tapped out for the day